operator/watchers/cilium_endpoint.go

```diff
@@ -163,3 +211,40 @@ func HasCE(ns, name string) (*cilium_api_v2.CiliumEndpoint, bool, error) {
 	cep := item.(*cilium_api_v2.CiliumEndpoint)
 	return cep, exists, nil
 }
+
+func endpointUpdated(cep *cilium_api_v2.CiliumEndpoint) {
+	if cebController != nil {
+		if cep.Status.Networking == nil || cep.Status.Identity == nil || cep.GetName() == "" || cep.Namespace == "" {
+			return
+		}
+		cebController.Manager.InsertCepInCache(k8s.ConvertCeptoCoreCep(cep))
+	}
+}
+
+func endpointDeleted(cep *cilium_api_v2.CiliumEndpoint) {
+	if cebController != nil {
```

Comment:
why do we need to check here? We would only call this function if CEB is enabled, hence cebController would always be instantiated when this function is called?

Reply:
Always good practice to check the pointer before accessing its fields.

Reply:
I don't think so...this just complicates code and makes reader wonder why it could be nil...(devil's advocate: using your logic, then we should check every single field access throughout the code).

Reply:
Checking for a valid pointer doesn't complicate code. In fact it improves code readability and reliability.

Reply:
In this particular case, the `nil` pointer check is already performed at L112 above. 

```
if cep := objToCiliumEndpoint(obj); cep != nil {
						endpointDeleted(cep)
					}
```

Reply:
@aditighag the check you pointed is for validating we received valid `ciliumendpoint' i.e. `cep`
However, `cebController` check at L225 is for checking for valid 'cebController`.

Reply:
I agree with @Weil0ng here, it makes me nervous to have this check here. If we have this check how are we sure we are not losing CEPs events because `cebController` was nil at a given point in time?